lib/srv/db/server.go

```diff
@@ -402,13 +403,17 @@ func (m *monitoredDatabases) isCloud(database types.Database) bool {
 	return false
 }
 
-func (m *monitoredDatabases) isDiscoveryResource(database types.Database) bool {
-	return database.Origin() == types.OriginCloud && m.isResource(database)
+// isDiscoveryResourceLocked returns whether a database was discovered by the
+// discovery service.
+// The lock must be held when calling this function.
+func (m *monitoredDatabases) isDiscoveryResourceLocked(database types.Database) bool {
+	return database.Origin() == types.OriginCloud && m.isResourceLocked(database)
 }
 
-func (m *monitoredDatabases) isResource(database types.Database) bool {
-	m.mu.RLock()
-	defer m.mu.RUnlock()
+// isResourceLocked returns whether a database is a dynamic database, aka a db
+// object.
+// The lock must be held when calling this function.
+func (m *monitoredDatabases) isResourceLocked(database types.Database) bool {
```

Comment:
Nit: isResourceLocked reads like it checks whether a resource is locked or not. Maybe use an underscore to indicate this is a function that is supposed to be called under a lock? `isResource_Locked`. Idk if this is a pattern we use elsewhere but that would be less confusing IMO. Same for others.